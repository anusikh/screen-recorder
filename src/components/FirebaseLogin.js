import React from "react";
import "../styles.css";
import { signInWithPopup } from "firebase/auth";
import { auth, provider } from "../firebase";
import logo from "../assets/logo.png";
import { LogContext } from "../context/LogContext";

function FirebaseLogin() {
  const [photoURL, setPhotoUrl] = React.useState("");

  const x = React.useContext(LogContext);
  const { loggedIn, dispatch } = x;

  const signInWithFirebase = () => {
    signInWithPopup(auth, provider)
      .then((res) => {
        const user = res.user;
        setPhotoUrl(user.photoURL);
        dispatch({ type: "change" });
      })
      .catch((error) => {
        console.log(error);
      });
  };

  return (
    <div className="FirebaseLogin__Main">
      {photoURL === "" ? (
        <img
          src={logo}
          width="60"
          height="60"
          style={{ margin: "0.4rem 0 0 0.4rem" }}
          onClick={signInWithFirebase}
        />
      ) : (
        <img src={photoURL} className="FirebaseLogin__image" alt="Login" />
      )}
    </div>
  );
}

export default FirebaseLogin;
