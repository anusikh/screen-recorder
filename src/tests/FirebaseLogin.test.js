import React from "react";
import renderer from "react-test-renderer";
import FirebaseLogin from "../components/FirebaseLogin";

test("FirebaseLogin Functionality Snapshot", () => {
  const component = renderer.create(<FirebaseLogin />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //   tree.props.onClick();
  console.log(tree.children[0].props.onClick);
  tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
