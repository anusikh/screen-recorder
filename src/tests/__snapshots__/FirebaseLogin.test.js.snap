// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirebaseLogin Functionality Snapshot 1`] = `
<div
  className="FirebaseLogin__Main"
>
  <button
    className="FirebaseLogin__Butt"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`FirebaseLogin Functionality Snapshot 2`] = `
<div
  className="FirebaseLogin__Main"
>
  <button
    className="FirebaseLogin__Butt"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
